client/reducers/getPreviousRoomNames-reducer.js

```diff
@@ -1,6 +1,13 @@
-const getPreviousRoomNameReducer = (state = ['The Grill', 'Taj Mahal', 'The High Castle', 'Your Mother\'s Bedroom'], action) => {
+const getPreviousRoomNameReducer = (state = [], action) => {
   switch (action.type) {
     case 'GET_PREVIOUS_ROOM_NAMES':
+      if (Array.isArray(action.rooms)) {
+        action.rooms.forEach((room) => {
+          state.push(room.roomname);
+        });
+      } else {
+        state.push(action.rooms);
+      }
       return state;
```

Comment:
This function is impure. Can we update it so that we return a new copy of the state with changes, rather than mutating the state directly?